scripts/pre-commit.php

```diff
@@ -80,15 +117,45 @@
     putenv('DBTEST=1');
 }
 
+// No php files, skip the php checks.
+if ($map['php'] === 0) {
+    putenv('SKIP_LINT_CHECK=1');
+    putenv('SKIP_STYLE_CHECK=1');
+}
+
+// If we have no php files and no OS' found then also skip unit checks.
+if ($map['php'] === 0 && empty($map['os'])) {
+    putenv('SKIP_UNIT_CHECK=1');
+}
+
+// If we have more than 4 (arbitrary number) of OS' then blank them out
+// Unit tests may take longer to run in a loop so fall back to all.
+if (count($map['os']) > 4) {
+    unset($map['os']);
+}
+
 // run tests in the order they were specified
+
 foreach (array_keys($options) as $opt) {
     $ret = 0;
     if ($opt == 'l' || $opt == 'lint') {
         $ret = run_check('lint', $passthru, $command_only);
     } elseif ($opt == 's' || $opt == 'style') {
         $ret = run_check('style', $passthru, $command_only);
     } elseif ($opt == 'u' || $opt == 'unit') {
-        $ret = run_check('unit', $passthru, $command_only, compact('fail_fast', 'os', 'module'));
+        if (!empty($map['os']) && $map['php'] === 0) {
+            foreach ($map['os'] as $os) {
```

Comment:
Instead of doing a loop, you can provide a regex in the os key:

```php
    if ($options['os']) {
        $phpunit_cmd .= " --group os --filter \"@{$options['os']}.*\"";
    }
```

Reply:
Well spotted. See if the update I've just done works.